static/js/containers/ChannelPage.js

```diff
@@ -16,41 +16,68 @@ import { getChannelName } from "../lib/util"
 
 import type { Dispatch } from "redux"
 import type { Match } from "react-router"
+import type { RestState } from "../flow/restTypes"
+import type { Channel, Post } from "../flow/discussionTypes"
 
 class ChannelPage extends React.Component {
   props: {
     match: Match,
     dispatch: Dispatch,
-    channels: Object,
-    postsForChannel: Object,
-    posts: Object
+    channels: RestState<Map<string, Channel>>,
+    postsForChannel: RestState<Map<string, Array<string>>>,
+    posts: RestState<Map<string, Post>>,
+    subscriptions: RestState<Array<Channel>>,
```

Comment:
I'll rename `subscribedChannels` but I'm not sure about `channel` or `channelDetail`. The redux state is the canonical collection of channels so `channels` seems to fit best there. Maybe `subscribedChannels` should contain the names for the channels and the channel objects should be stored in `channels`, similar to how the post objects are stored in `posts` and the ids for each post are stored in `postsForChannel`